restorer.go

```diff
@@ -4,33 +4,42 @@ import (
 	"github.com/pkg/errors"
 	"golang.org/x/sync/errgroup"
 
+	"github.com/buildpacks/lifecycle/api"
 	"github.com/buildpacks/lifecycle/buildpack"
+	"github.com/buildpacks/lifecycle/cmd"
 	"github.com/buildpacks/lifecycle/layers"
 	"github.com/buildpacks/lifecycle/platform"
 )
 
 type Restorer struct {
-	LayersDir  string
-	Buildpacks []buildpack.GroupBuildpack
-	Logger     Logger
+	LayersDir              string
+	Buildpacks             []buildpack.GroupBuildpack
+	Logger                 Logger
+	SkipLayers             bool
+	LayerAnalyzer          LayerAnalyzer
+	CacheMetadataRetriever CacheMetadataRetriever
+	Platform               cmd.Platform
+	LayersMetadata         platform.LayersMetadata
 }
 
-// Restore attempts to restore layer data for cache=true layers, removing the layer when unsuccessful.
-// If a usable cache is not provided, Restore will remove all cache=true layer metadata.
+// Restore restores metadata for launch and cache layers into the layers directory and attempts to restore layer data for cache=true layers, removing the layer when unsuccessful.
```

Comment:
I wonder if it's worth describing the per-API behavior in this comment.

Reply:
See https://github.com/buildpacks/lifecycle/pull/582#discussion_r614942414